hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,853 @@
+                      ] + common_parameters
+
+        # test for private resource
+        self.assertFalse(self.res.raccess.public)
+        self.assertFalse(self.res.raccess.discoverable)
+
+        self._run_tests(self.request, parameters)
+
+        # test for discoverable resource
+        self.assertFalse(self.res.raccess.discoverable)
+        self.res.raccess.discoverable = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for public resource
+        self.assertTrue(self.res.raccess.discoverable)
+        self.assertFalse(self.res.raccess.public)
+        self.res.raccess.discoverable = False
+        self.res.raccess.public = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for immutable/published resource
+        self.assertFalse(self.res.raccess.immutable)
+        self.assertFalse(self.res.raccess.published)
+        self.assertTrue(self.res.raccess.public)
+
+        self.res.raccess.immutable = True
+        self.res.raccess.published = True
+        self.res.raccess.save()
+
+        parameters = [
+                      # super user has authorization for setting resource flags (e.g., public,
+                      # published/immutable, shareable etc) for a resource that has been published
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.SET_RESOURCE_FLAG,
+                       'success': True, 'exception': None},
+                      # super user has no authorization for deleting a published resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.DELETE_RESOURCE,
+                       'success': False, 'exception': PermissionDenied}
+                      ] + common_parameters
+
+        self._run_tests(self.request, parameters)
+
+    def test_authorize_anonymous_user(self):
+        self.request.user = AnonymousUser()
+        common_parameters = [
+                      # anonymous user has no authorization for resource edit (metadata and content files) for a
+                      # private resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.EDIT_RESOURCE,
+                       'success': False, 'exception': PermissionDenied},
+                      # anonymous user has no authorization for deleting a private resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.DELETE_RESOURCE,
+                       'success': False, 'exception': PermissionDenied},
+                      # anonymous user has no authorization for setting resource flags for a private resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.SET_RESOURCE_FLAG,
+                       'success': False, 'exception': PermissionDenied}
+                      ]
```

Comment:
@pkdash Should add permission parameter for testing Action_To_Authorize.CREATE_RESOURCE_VERSION action to the list for anonymous users who are not allowed to version a resource